File: JavaScript/leetcode75/src/main/Problem20_FindTheDifferenceOfTwoArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[][]}
 */
var findDifference = function (nums1, nums2) {
  let set1 = new Set(nums1);
  let set2 = new Set(nums2);

  for (let num of set1) {
    if (set2.has(num)) {
      set1.delete(num);
      set2.delete(num);
    }
  }

  return [[...set1], [...set2]];
};

module.exports = findDifference;

File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem20_FindTheDifferenceOfTwoArrays.java
package org.phoenix.leetcode.challenges;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Problem20_FindTheDifferenceOfTwoArrays {
    public List<List<Integer>> findDifference(int[] nums1, int[] nums2) {
        Set<Integer> set1 = new HashSet<>();
        Set<Integer> set2 = new HashSet<>();
        List<List<Integer>> diffList = new ArrayList<>();
        diffList.add(new ArrayList<>());
        diffList.add(new ArrayList<>());

        for (int num : nums1) {
            set1.add(num);
        }
        for (int num : nums2) {
            set2.add(num);
        }

        for (int num : set1) {
            if (!set2.contains(num)) {
                diffList.get(0).add(num);
            }
        }
        for (int num : set2) {
            if (!set1.contains(num)) {
                diffList.get(1).add(num);
            }
        }
        return diffList;
    }
}
